### PR TITLE
Split routes.rb in smaller files (webui/api)

### PR DESCRIPTION
**Pull request by vpereira**
To make easier to navigate through the routes files, split it in two
domains:
  - webui (config/routes/webui_routes.rb)
  - api (config/routes/api_routes.rb)

There are some room for improvement, like I would like to have a base constrain with the bare minimum contrains that apply for webui and api and go specifics in the respective files.. We can do it however in a new iteration.

### CODE REVIEW

File: src/api/config/routes/routes_helper.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
+
```
Comment (dmarcoux) [inline src/api/config/routes/routes_helper.rb]: This line should be removed.

File: src/api/config/routes/api_routes.rb [hunk 2]
```diff
@@ -0,0 +1,408 @@
...
+    get 'issue_trackers/issues_in' => 'issue_trackers#issues_in'
+    resources :issue_trackers, only: [:index, :show, :create, :update, :destroy] do
+      resources :issues, only: [:show] # Nested route
+    end
+
+    ### /tag
+
+    # routes for tagging support
+    #
+    # get 'tag/_all' => 'tag',
```
Comment (dmarcoux) [inline src/api/config/routes/api_routes.rb]: Should we remove those commented routes? I guess it's fine to do it.

File: dist/obs-server.spec [hunk 3]
```diff
@@ -831,6 +831,9 @@ usermod -a -G docker obsservicerun
 
 /srv/www/obs/api/config/boot.rb
 /srv/www/obs/api/config/routes.rb
+/srv/www/obs/api/config/routes/api_routes.rb
```
Comment (hennevogel) [inline dist/obs-server.spec]: You should add the whole directory, not just the files in it.

Comment (adrianschroeter) [reply dist/obs-server.spec]: nah, it must not overwrite %config definitions for other files inside

Comment (coolo) [reply dist/obs-server.spec]: I don't think config/routes should be %confg

File: src/api/config/routes/api_routes.rb (hunk 2 above)
Comment (vpereira) [reply src/api/config/routes/api_routes.rb]: I would remove it.

File: src/api/config/routes/routes_helper.rb (hunk 1 above)
Comment (vpereira) [reply src/api/config/routes/routes_helper.rb]: done

File: src/api/config/routes/api_routes.rb (hunk 2 above)
Comment (dmarcoux) [reply src/api/config/routes/api_routes.rb]: Yes, I'm fine with this!

File: dist/obs-server.spec (hunk 3 above)
Comment (vpereira) [reply dist/obs-server.spec]: done @hennevogel , looks package is building successfully.

**Comment by hennevogel** [comment]
@openSUSE/obs-frontend any other comments about this?

**Pull request merged**